src/query/graphite/native/builtin_functions.go

```diff
@@ -94,6 +95,46 @@ func sortByMaxima(ctx *common.Context, series singlePathSpec) (ts.SeriesList, er
 	return highestMax(ctx, series, len(series.Values))
 }
 
+/*
+Compares the maximum of each series against the given `value`. If the series
+maximum is greater than `value`, the regular expression search and replace is
+applied against the series name to plot a related metric
+
+e.g. given useSeriesAbove(ganglia.metric1.reqs,10,'reqs','time'),
+the response time metric will be plotted only when the maximum value of the
+corresponding request/s metric is > 10
+
+&target=useSeriesAbove(ganglia.metric1.reqs,10,"reqs","time")
+*/
+func useSeriesAbove(ctx *common.Context, seriesList singlePathSpec , maxAllowedValue float64, search, replace string) (ts.SeriesList, error) {
+	results := make([]*ts.Series, 0, len(seriesList.Values))
+	for _, series := range seriesList.Values {
+		if series.SafeMax() > maxAllowedValue {
+			opts := storage.FetchOptions{
+				StartTime: ctx.StartTime,
+				EndTime:   ctx.EndTime,
+				DataOptions: storage.DataOptions{
+					Timeout: ctx.Timeout,
+				},
+			}
+			seriesName := strings.Replace(series.Name(), search, replace, 1)
+			result, err := ctx.Engine.FetchByQuery(ctx, seriesName, opts)
+			if err != nil {
+				return ts.NewSeriesList(), err
+			}
+			if len(result.SeriesList) > 0 {
+				results = append(results, result.SeriesList[0])
```

Comment:
Should this only take the first series found or all series?

Reply:
This was bad/hacky/wrong implementation, and I have fixed it.